src/vs/workbench/services/url/electron-browser/urlService.ts

```diff
@@ -28,6 +30,19 @@ export class RelayURLService extends URLService implements IURLHandler {
 		openerService.registerOpener(this);
 	}
 
+	create(options?: Partial<UriComponents>): URI {
+		const uri = super.create(options);
+
+		let query = uri.query;
+		if (!query) {
+			query = `windowId=${this.windowService.windowId}`;
+		} else {
+			query += `&windowId=${this.windowService.windowId}`;
```

Comment:
@bpasero This should be wrapped by `encodeURIComponent`, especially because window ids contain `:` characters.